packages/api/src/routes/internal/index.ts

```diff
@@ -340,10 +341,47 @@ router.post(
   })
 );
 
+/**
+ * PATCH /internal/cx-mapping/update-secondary-mappings
+ *
+ * Update secondary mapping in a cx mapping
+ *
+ * @param req.query.cxId - The cutomer's ID.
+ * @param req.query.id - The cx mapping ID.
+ * @param req.query.source - the mapping source
+ */
+router.patch(
+  "/cx-mapping/update-secondary-mappings",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getUUIDFrom("query", req, "cxId").orFail();
+    const id = getFrom("query").orFail("id", req);
+    const source = getFromQueryOrFail("source", req);
+    if (!isCxMappingSource(source)) {
+      throw new BadRequestError(`Invalid source for cx mapping`, undefined, { source });
```

Comment:
Just for future reference, in other cases it can be useful to pass all the context of an error in the third parameter which currently only has `{ source }` - so in this case, might also pass it as `{ cxId, id, source }`. In this case, it's an internal endpoint that doesn't lean on Sentry logging, so NBD.